src/main/java/ognl/DefaultClassResolver.java

```diff
@@ -41,28 +42,27 @@
  */
 public class DefaultClassResolver extends Object implements ClassResolver
 {
-    private Map     classes = new HashMap(101);
+    private final Map<String, Class> classes = new ConcurrentHashMap<>(101);
 
     public DefaultClassResolver()
     {
         super();
     }
 
-	public Class classForName(String className, Map context) throws ClassNotFoundException
-	{
-	    Class       result = null;
-
-        if ((result = (Class)classes.get(className)) == null) {
-    		try {
-    		    result = Class.forName(className);
-    		} catch (ClassNotFoundException ex) {
-    			if (className.indexOf('.') == -1) {
-    			    result = Class.forName("java.lang." + className);
-        			classes.put("java.lang." + className, result);
-        		}
-    		}
-			classes.put(className, result);
-    	}
-	    return result;
-	}
+    public Class classForName(String className, Map context) throws ClassNotFoundException
+    {
+        Class result = classes.get(className);
+        if (result != null) {
+            return result;
+        }
+        if (className.indexOf('.') == -1) {
+            String langClassName = "java.lang." + className;
+            result = Class.forName(langClassName);
+            classes.put(langClassName, result);
+        } else {
+            result = Class.forName(className);
+        }
+        classes.put(className, result);
```

Comment:
This will duplicate `langClassName`, so we will end up with two entries for the same class.

Reply:
Thanks for your comment.
I think this behavior is same as the current version. Should change this behavior in this change?

Reply:
Right, which means the `java.lang.` classes will be always located by the short version (`Integer` instead of `java.lang.Integer`). I think `.put` in line 61 can be thrown aways.

Reply:
I've fixed your comment. please review again.
Thanks.